plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/controls/querylog/QueryLogViewer.java

```diff
@@ -1136,14 +1132,37 @@ protected void buttonPressed(int buttonId) {
         @Override
         protected SQLDialect getSQLDialect() {
             if (object.getObject() instanceof QMMStatementExecuteInfo) {
-                SQLDialect dialect = ((QMMStatementExecuteInfo) object.getObject()).getStatement().getConnection().getSQLDialect();
-                if (dialect != null) {
-                    return dialect;
+                var executeInfo = (QMMStatementExecuteInfo) object.getObject();
+                var container = getDataSourceContainer(executeInfo);
+                var sqlDialect = getSqlDialectFromContainer(container);
+                if (getSqlDialectFromContainer(container) != null) {
+                    return sqlDialect;
                 }
             }
             return super.getSQLDialect();
         }
 
+        @Nullable
+        private SQLDialect getSqlDialectFromContainer(DBPDataSourceContainer container) {
+            if (container == null) {
+                return null;
+            }
+            var dataSource = container.getDataSource();
+            if (dataSource != null) {
+                return container.getDataSource().getSQLDialect();
+            }
+            DBPDriver driver = DataSourceProviderRegistry.getInstance().findDriver(container.getDriver().getId());
+            if (driver == null) {
+                return null;
+            }
+            driver = driver.createOriginalCopy();
+            try {
+                return driver.createOriginalCopy().getScriptDialect().createInstance();
```

Comment:
Why do we need `createOriginalCopy` here?

Reply:
fixed